scipy/stats/mstats_basic.py

```diff
@@ -642,19 +642,9 @@ def linregress(*args):
     replaced by the non-masked docstring + some info on missing data.
 
     """
-    if len(args) == 1:
-        # Input is a single 2-D array containing x and y
-        args = ma.array(args[0], copy=True)
-        if len(args) == 2:
-            x = args[0]
-            y = args[1]
-        else:
-            x = args[:, 0]
-            y = args[:, 1]
-    else:
-        # Input is two 1-D arrays
-        x = ma.array(args[0]).flatten()
-        y = ma.array(args[1]).flatten()
+    x = ma.array(x).flatten()
```

Comment:
This doesn't look right for a single 2-D input array. It should be split in two, like the old code did. Can you add a test for this and undo the change here?